src/sentry/snuba/entity_subscription.py

```diff
@@ -301,6 +301,9 @@ def __init__(
             )
         self.org_id = extra_fields["org_id"]
         self.time_window = time_window
+        # If we are using any metrics-based dataset, we will enable the use of metrics layer and also the snql
+        # generation through it.
+        self.use_metrics_layer = self.dataset in {Dataset.Metrics, Dataset.PerformanceMetrics}
```

Comment:
Nit: Since all subclasses of `BaseMetricsEntitySubscription` use one of these two datasets, not totally sure if this is necessary?

Could simplify a lot of the code below though.

Reply:
The goal of my changes was to be as explicit as possible with the code but I can for sure simplify some parts.

Reply:
I will iterate on some improvements now, since you reviewed the general implementation. Will update the PR!

Reply:
Btw, my ideas of introducing this boolean was to have both implementations that can be switched in case of issues or need. If we decide to avoid completely having a flag, I could simplify the code and move everything to metrics layer.